File: Java/Recursion-1/changeXY.java

public class recursion1 {
	
	// given a string, compute recursively (no loops) a new string where all the lowercase 'x' chars have been changed to 'y' chars
	public String changeXY(String str) {
		
		if (str.isEmpty()) {
			return "";
		}

		if (str.charAt(0) != 'x') {
			return str.charAt(0) + changeXY(str.substring(1));
		}
		else {
			return 'y' + changeXY(str.substring(1));
		} 
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.changeXY("codex"));
		System.out.println(o.changeXY("xxhixx"));
		System.out.println(o.changeXY("xhixhix"));
		
	}

}

File: JavaScript/Recursion-1/changeXY.js

// given a string, compute recursively a new string where all the lowercase 'x' chars have been changed to 'y' chars
const changeXY = (str) => {
    if (!str) {
        return '';
    }

    return str[0] === 'x' ? 'y' + changeXY(str.substring(1)) : str[0] + changeXY(str.substring(1));
};

console.log(changeXY('codex'));
console.log(changeXY('xxhixx'));
console.log(changeXY('xhixhix'));
